dask_sql/physical/rel/logical/offset.py

```diff
@@ -0,0 +1,113 @@
+from typing import TYPE_CHECKING
+
+import dask.dataframe as dd
+
+from dask_sql.datacontainer import DataContainer
+from dask_sql.physical.rel.base import BaseRelPlugin
+from dask_sql.physical.rex import RexConverter
+from dask_sql.physical.utils.map import map_on_partition_index
+
+if TYPE_CHECKING:
+    import dask_sql
+    from dask_planner.rust import LogicalPlan
+
+
+class DaskOffsetPlugin(BaseRelPlugin):
+    """
+    Offset is used to modify the effective expression bounds in a larger table
+    (OFFSET).
+    """
+
+    class_name = "Offset"
+
+    def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContainer:
+        (dc,) = self.assert_inputs(rel, 1, context)
+        df = dc.df
+        cc = dc.column_container
+
+        offset_node = rel.offset()
+
+        offset = offset_node.getOffset()
+        if offset:
+            offset = RexConverter.convert(rel, offset, df, context=context)
+
+        # TODO: `Fetch` is not currently supported
+        # end = offset_node.getFetch()
+        # if end:
+        #     end = RexConverter.convert(rel, end, df, context=context)
+
+        #     if offset:
+        #         end += offset
+        end = df.shape[0].compute()
+        print(f"End Size: {end} other: {len(df)}")
+
+        df = self._apply_offset(df, offset, end)
+
+        print(f"Size of DF: {df.shape[0].compute()} after applying Offset: {offset}")
+
+        cc = self.fix_column_to_row_type(cc, rel.getRowType())
+        # No column type has changed, so no need to cast again
+        return DataContainer(df, cc)
+
+    def _apply_offset(self, df: dd.DataFrame, offset: int, end: int) -> dd.DataFrame:
```

Comment:
Clarifying question - if we do something like `SELECT ... LIMIT 100 OFFSET 50`, would that end up calling the offset plugin and then the limit plugin immediately after? Or is the offset plugin for any `LIMIT` statements with `OFFSETs`, and the limit plugin is just for plain `LIMITs`?

If the former, wondering if we can get away with using something like `tail` to do this relatively simply, since an `OFFSET` alone is essentially just a backwards `LIMIT`. If the latter, I would reccommend incorporating the changes made in https://github.com/dask-contrib/dask-sql/pull/517 here, which should improve things by switching over to `map_partitions` (note that this allows us to drop `map_on_partition_index`).

Reply:
It is the first case. 

First the `offset.py` plugin would be invoked and then the `limit.py` plugin. DataFusion treats offset and limit as different logical operators where Calcite treated them as the "same". Honestly, both could still be in the same file but it seemed more clear to have the logic split apart in separate files to more closely mirror the actual `LogicalPlan` data structure from DataFusion. Could be changed realistically though at the expense of making the logic less clear.

I'm not very familiar with tail but that sounds like something which could be used. Mind posting a code snippet or commit with those changes you are talking about?